File: labs-training-master/java/solid/solid-java-1/src/main/java/es/emergya/labs/training/solid_java_1/srp/excercise_00/CarManager.java
package es.emergya.labs.training.solid_java_1.srp.excercise_00;

//import java.util.Arrays;
import java.util.List;

public class CarManager {

	public CarRepository carRepository;

	public CarManager(){
		this.carRepository = new CarRepository();
	}


	public Car getCarById(final String id) {
		return this.carRepository.getCarById(id);
	}

	public String getCarsNames() {
		
		StringBuilder sb = new StringBuilder();
		List<Car> databases = carRepository.getCars();
		
		for (Car car : databases) {
			
			sb.append(car.getBrand());
			sb.append(" ");
			sb.append(car.getModel());
			sb.append(", ");
		}
		
		return sb.substring(0, sb.length() - 2);
	}

	public Car getBestCar() {
		
		Car bestCar = null;

		List<Car> databases = carRepository.getCars();
		for (Car car : databases) {
			
			if (bestCar == null || car.getModel().compareTo(bestCar.getModel()) > 0) {
				
				bestCar = car;
			}
		}
		
		return bestCar;
	}

}

File: labs-training-master/SolidPythonResuelto/main/srp/CarManager.py
from Car import Car
from typing import Optional
from CarRepository import CarRepository


class CarManager :
    def __init__(self) -> None:
        self.carRepository = CarRepository()
    def  getCarById(self, id : str) -> Optional[Car]:
        self.carRepository.getCarById(id)
    def  getCarsNames(self) -> 'list[str]' :
        sb: 'list[str]' = []
        for car in self.carRepository.getCars :
            sb.append(car.getBrand() + " " + car.getModel() + ", ")
        return  sb[:2]
    def  getBestCar(self) -> Car :
        bestCar : Car = None
        for car in self.carRepository.getCars :
            if (bestCar == None or car.getModel() < bestCar.getModel()) :
                bestCar = car
        return  bestCar